spec/lib/parameter_substitution/expression_spec.rb

```diff
@@ -57,5 +57,19 @@ def parse_expression(parameter_expression, context_options = {})
         expect("red1").to eq(parse_expression("<color><integer>", destination_encoding: :raw).evaluate)
       end
     end
+
+    context "#substitution_parameter_names" do
+      it "return expression list parameter names" do
+        expression = parse_expression("<simple_text>")
+        expect(expression.substitution_parameter_names).to eq(["simple_text"])
+      end
+    end
+
+    context "#method_names" do
+      it "return expression list method names" do
+        expression = parse_expression("<simple_text.do_a_barrel_roll>")
+        expect(expression.method_names).to eq(["do_a_barrel_roll"])
+      end
+    end
```

Comment:
Since we only have 1 test for each of these, let's make sure it can handle >1 expression in a string.